sklearn/preprocessing/_discretization.py

```diff
@@ -198,7 +198,7 @@ def __init__(
         self.random_state = random_state
 
     @_fit_context(prefer_skip_nested_validation=True)
-    def fit(self, X, y=None, sample_weight=None):
+    def fit(self, X, y=None, sample_weight=None, quantile_method="warn"):
```

Comment:
I missed this in my previous review, but there should not be any new parameter in the `fit` method.

```suggestion
    def fit(self, X, y=None, sample_weight=None):
```

It should be a constructor parameter for `__init__` (as we do for `strategy="quantile"`).